recurring_contract/models/product_pricelist_item.py

```diff
@@ -0,0 +1,28 @@
+from odoo import models, _
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = 'product.pricelist.item'
+
+    def update_cl_amount(self):
+        for item in self:
+            contract_lines = self.env['recurring.contract.line'].search([
+                ('contract_id.pricelist_id', '=', item.pricelist_id.id),
+                ('product_id', '=', item.product_id.id),
+                ('contract_id.state', '=', ['waiting', 'active'])
+            ])
+            for cl in contract_lines:
+                price = cl.contract_id.pricelist_id.get_product_price(cl.product_id,
+                                                                      cl.quantity,
+                                                                      cl.contract_id.partner_id)
+                if cl.amount != price:
+                    cl.amount = price
+        return {
```

Comment:
As the process will be run in the background, I don't think anybody will get this notification. Unless some people will run the method manually somewhere in the user interface.

Reply:
It's for the future :) In case we want this update to be called manually we have a feedback of what happened :D